source/slides/05_linux_basics.rst

```diff
@@ -0,0 +1,299 @@
+  Logical Extent (LE), Chunks of data in a LV with the same size
+
+`LVM Howto`_ (a bit dated, but still relevant)
+
+.. _LVM Howto: http://tldp.org/HOWTO/LVM-HOWTO/
+
+LVM Example
+-----------
+
+.. rst-class:: build
+
+#. Two disk partitions: sda1 (800MB) & sdb1 (400MB)
+#. Format each partition into a PV
+#. The PV's are divided up into 4MB chunks (PE): sda1 (200PE) / sdb1 (100PE)
+#. Create a VG using sda1 and sda2
+#. Create an LV of size 1GB (uses linear mapping by default)
+#. Total of 250 LE's are required
+#. 200PE's are used in sda1, 50PE's are used in sdb1, 50PE are free
+#. Create a filesystem from the LV and mount it
+#. PROFIT!
+
+LVM Visualized
+--------------
+
+.. insert graphic here
+
+LVM commands
+------------
+
+.. code-block:: bash
+
+  $ yum install lvm2
+
+  $ pvcreate /dev/loop1 /dev/loop2
+    Physical volume "/dev/loop1" successfully created
+    Physical volume "/dev/loop2" successfully created
+
+  $ vgcreate lvm /dev/loop1 /dev/loop2
+    Volume group "lvm" successfully created
+
+  $ lvcreate -n home -L 1G lvm
+    Logical volume "home" created
+
+  $ mkfs.ext4 /dev/lvm/home
+  $ mount /dev/lvm/home /mnt
+
+LVM commands
+------------
+
+All commands start with ``pv``, ``vg`` or ``lv``
+
+::
+
+  pvchange   pvcreate   pvmove     pvresize   pvscan
+  pvck       pvdisplay  pvremove   pvs
+
+  vgcfgbackup    vgconvert      vgextend       vgmknodes      vgs
+  vgcfgrestore   vgcreate       vgimport       vgreduce       vgscan
+  vgchange       vgdisplay      vgimportclone  vgremove       vgsplit
+  vgck           vgexport       vgmerge        vgrename
+
+  lvchange     lvextend     lvmdiskscan  lvmsar       lvresize
+  lvconvert    lvm          lvmdump      lvreduce     lvs
+  lvcreate     lvmchange    lvmetad      lvremove     lvscan
+  lvdisplay    lvmconf      lvmsadc      lvrename
+
+Useful tricks with LVM
+----------------------
+
+* Move LE's between PV's (linear vs. striped problem)
+* Move from one PV to another PV live
+* Rename VG's or LV's live
+* Reduce, remove or add LV's as needed
+* Make atomic snapshots for backups
+
+Issues with LVM
+---------------
+
+* Requires an initrd if the rootfs is in lvm
+
+  * Can be make troubleshooting difficult sometimes
+
+* Can't move LV's between VG's
+* If a disk dies, all the PE's are gone (unless RAID is used in LVM)
+* Using LVM inside of LVM can be hard to troubleshoot
+
+  * Ganeti VMs
+
+RAID
+-----
+
+**Software (mdadm or LVM)**
+
+* sda1/sda2 -> md0 -> PV -> VG -> LV - > filesystem
+* sda1/sda2 -> PV -> VG -> -LV (raid#) -> filesystem
+* mdadm + lvm usually preferred
+* Performance has improved quite a bit over the years
+
+**Hardware**
+
+* Dedicated hardware to RAID
+* Prevent boot problems that mdadm may have
+* Proprietary tools
+* Claim better performance
+
+RAID Levels
+-----------
+
+.. csv-table::
+  :header: Level, Minimum disks, Description
+
+  RAID0, 2, Striping
+  RAID1, 2, Mirroring
+  RAID5, 3, Block-level striping with distributed parity
```

Comment:
Well, I basically mention that below... Unless you're wanting me to reword it. I was trying to keep it simple.